KhivaCsharp/KhivaCsharp/Matrix.cs

```diff
@@ -0,0 +1,135 @@
+                index.Reference = indexReference;
+                var tuple = Tuple.Create(KhivaArray.Create(discords_distances),
+                             KhivaArray.Create(discords_indices),
+                             KhivaArray.Create(subsequence_indices));
+                return tuple;
+            }
+           
+            /// <summary>
+            /// Primitive of the findBestNMotifs function.
+            /// </summary>
+            /// <param name="profile">The matrix profile containing the minimum distance of each subsequence.</param>
+            /// <param name="index">The matrix profile index containing where each minimum occurs.</param>
+            /// <param name="m">Subsequence length value used to calculate the input matrix profile.</param>
+            /// <param name="n">Number of motifs to extract.</param>
+            /// <param name="self_join">Indicates whether the input profile comes from a self join operation or not. It determines
+            /// whether the mirror similar region is included in the output or not.</param>
+            /// <returns>Tuple with 
+            /// the distance of the best N motifs, 
+            /// the indices of the best N motifs, 
+            /// the indices of the query sequences that produced
+            /// the minimum reported in the motifs.</returns>
+            public static Tuple<KhivaArray, KhivaArray, KhivaArray> FindbestNMotifs(KhivaArray profile, KhivaArray index, long m, long n, bool self_join=false)
+            {
+                IntPtr profileReference = profile.Reference;
+                IntPtr indexReference = index.Reference;
+                IntPtr motif_distances; IntPtr motif_indices; IntPtr subsequence_indices;
+                interop.DLLMatrix.find_best_n_motifs(ref profileReference, ref indexReference, ref m, ref n,
+                                                        out motif_distances, out motif_indices, out subsequence_indices,
+                                                        ref self_join);
+                profile.Reference = profileReference;
+                index.Reference = indexReference;
+                var tuple = Tuple.Create(KhivaArray.Create(motif_distances),
+                                         KhivaArray.Create(motif_indices),
+                                         KhivaArray.Create(subsequence_indices));
+                return tuple;
+            }
+
+            /// <summary>
+            /// Primitive of the STOMP algorithm.
+            /// 
+            /// [1] Yan Zhu, Zachary Zimmerman, Nader Shakibay Senobari, Chin-Chia Michael Yeh, Gareth Funning, Abdullah Mueen,
+            /// Philip Brisk and Eamonn Keogh (2016). Matrix Profile II: Exploiting a Novel Algorithm and GPUs to break the one
+            /// Hundred Million Barrier for Time Series Motifs and Joins. IEEE ICDM 2016.
+            /// </summary>
+            /// <param name="tssa">Query time series</param>
```

Comment:
missing dots at the end of the l¡next lines:

            /// <param name="tssa">Query time series</param>
            /// <param name="tssb">Reference time series</param>